src/async_impl/request.rs

```diff
@@ -40,6 +42,19 @@ impl Request {
             url,
             headers: HeaderMap::new(),
             body: None,
+            timeout: None
+        }
+    }
+
+    /// Constructs a new request with timeout.
+    #[inline]
+    pub fn new_with_timeout(method: Method, url: Url, timeout: Duration) -> Self {
```

Comment:
I'd leave the new constructor out, users can use the builder or the `timeout_mut` method.